nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -307,12 +320,18 @@ namespace, getRef().getName()),
 
   public void renameTable(TableIdentifier from, TableIdentifier to) {
     getRef().checkMutable();
+    try {
+      getRef().refresh(api);
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(e);
+    }
 
-    IcebergTable existingFromTable = table(from);
+    UpdateableReference updateableReference = getRef();
+    IcebergTable existingFromTable = table(from, updateableReference.getReference());
```

Comment:
What is the point of allowing an explicit hash here and below? Shouldn't it fail, if an explicit hash is provided?

Reply:
I get it now. rename and drop table doesn't have to worry about table state. Reverted it.